pkg/network/node/subnets.go

```diff
@@ -148,7 +149,11 @@ func (node *OsdnNode) getLocalSubnet() (string, error) {
 		subnet, err = node.networkClient.NetworkV1().HostSubnets().Get(context.TODO(), node.hostName, metav1.GetOptions{})
 		if err == nil {
 			if err = common.ValidateHostSubnet(subnet); err != nil {
-				return false, err
+				// fail only if the system-maintained fields of hostsubnet are invalid.
+				if strings.Contains(err.Error(), "host") || strings.Contains(err.Error(), "hostIP") || strings.Contains(err.Error(), "subnet") {
```

Comment:
no... that's too fragile.

It's probably best to just make `ValidateHostSubnet` not validate the egress IP fields; the egress IP code doesn't use `ValidateHostSubnet` anyway, it does its own checking.

Hm... actually it mostly _doesn't_ do any checking. That needs to be fixed (maybe as a separate PR).

Reply:
ok yea, actually for me this whole `main hostsubnet watcher` and then the `egressip hostsubnet watcher` concepts are confusing (for me).

So here's what I'll do. I'll separate the validation functions into `ValidateHostSubnet` and `ValidateHostSubnetEgress`. I'll use the former in the `hostSubnetWatcher` and latter in the `EgressIPTracker` portions.